lib/smart_proxy_host_reports/puppet_processor.rb

```diff
@@ -101,14 +101,14 @@ def build_report
     if debug_payload?
       logger.debug { JSON.pretty_generate(@body) }
     end
-    report = build_report_root(
+    build_report_root(
       format: "puppet",
       version: 1,
       host: @body["host"],
       reported_at: @body["reported_at"],
       status: 0,
       proxy: @body["proxy"],
-      body: @body,
+      body: @body.to_json,
```

Comment:
It looks like there is a TODO in `build_report_root` that hints at this. Is it something that should be done in that method?

Reply:
Yeah this was not implemented yet:

```
    # TODO add body string serialization and metric with total time
    # and for tests there must be an option to turn this off
```

My thinking was that JSON in JSON is very hard to read so I was thinking to have some flag so in tests we could generate it without JSON in string so fixtures are easy to read and compare. This would need to be HTTP parameter so it can be used in integration tests.

Could you implement this?